src/main/java/edu/hm/hafner/coverage/parser/OpenCoverParser.java

```diff
@@ -0,0 +1,436 @@
+    private static final QName UID = new QName("uid");
+    private static final QName FULL_PATH = new QName("fullPath");
+    private static final QName METHOD_INTRUCTION_COVERED = new QName("visitedSequencePoints");
+    private static final QName METHOD_INTRUCTION_TOTAL = new QName("numSequencePoints");
+    private static final QName METHOD_BRANCH_COVERED = new QName("visitedBranchPoints");
+    private static final QName METHOD_BRANCH_TOTAL = new QName("numBranchPoints");
+    private static final QName METHOD_CYCLOMATIC_COMPLEXITY = new QName("cyclomaticComplexity");
+
+    @Override
+    protected ModuleNode parseReport(final Reader reader, final FilteredLog log) {
+        try {
+            var eventReader = new SecureXmlParserFactory().createXmlEventReader(reader);
+            var root = new ModuleNode("-");
+            var found = false;
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+                if (event.isStartElement()) {
+                    var startElement = event.asStartElement();
+                    var tagName = startElement.getName();
+                    if (MODULE.equals(tagName) && startElement.getAttributeByName(MODULE_SKIPPED) == null) {
+                        readPackage(eventReader, root);
+                        found = true;
+                    }
+                }
+            }
+            if (found) {
+                return root;
+            }
+            throw new NoSuchElementException("No coverage information found in the specified file.");
+        }
+        catch (XMLStreamException exception) {
+            throw new ParsingException(exception);
+        }
+    }
+
+    @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.CognitiveComplexity"})
+    private PackageNode readPackage(final XMLEventReader reader, final ModuleNode root) throws XMLStreamException {
+        Map<String, String> files = new LinkedHashMap<>();
+        List<CoverageClassHolder> classes = new LinkedList<>();
+        PackageNode packageNode = null;
+        while (reader.hasNext()) {
+            XMLEvent event = reader.nextEvent();
+            if (event.isStartElement()) {
+                var nextElement = event.asStartElement();
+                if (CLASS.equals(nextElement.getName())) {
+                    classes.add(readClass(reader));
+                }
+                else if (FILE.equals(nextElement.getName())) {
+                    var fileName = getValueOf(nextElement, FULL_PATH);
+                    var uid = getValueOf(nextElement, UID);
+                    var relativePath = PATH_UTIL.getRelativePath(fileName);
+                    files.put(uid, relativePath);
+                }
+                else if (MODULE_NAME.equals(nextElement.getName())) {
+                    String packageName = reader.nextEvent().asCharacters().getData();
+                    packageNode = root.findOrCreatePackageNode(packageName);
```

Comment:
On .NET side it's a module (like a project) that is part of part of a solution. The closest equivalence of package are namespace which is quiet different from Java.

I'm not sure what's best on the coverage-model if we need to threat .NET module as package or as coverage module.

Reply:
I would prefer to use module. I am using packages as synonym for namespaces in the coverage plugin and in the warnings plugin.  Normally a hierarchy looks like
 
     module
/                     \
package 1   package 2

or 

     module
/                     \
submodule 1   submodule 2
| | | |
packages

The idea is to show packages (or namespaces) side by side with the file names but modules are grouped somewhat different (not implemented yet).

Reply:
Okay, that's fine for me.

I can link all classed of a module under a "-" package

Reply:
If you don't have namespaces (or packages) you can omit them, that should work fine. In the UI packages are only shown if there is free space available.

(I do have code to extract namespaces from source files in the warnings plugin. Maybe it makes sense to reuse that code in the coverage plugin as well?)

Reply:
I was checking the implementation of the opencover adapter (I'm not the original developer of it) and there was not any notion of module or package/namespace (Expect the "Total coverage" I think all other views are broken)

Perhaps it could be an improvement of an other PR to extract directly from source file.

![opencover_plugin](https://github.com/jenkinsci/coverage-model/assets/825750/5e1c5565-d0ca-4822-8327-7f51f68703ac)

Now that I fixed the codecov I only remain to change the tree to not create package but submodule.

Thanks again for the review, was not a very easy PR ;)